src/nvim/api/buffer.c

```diff
@@ -179,6 +179,11 @@ void buffer_set_line_slice(Buffer buffer,
     return;
   }
 
+  if (!inbounds(buf, start)) {
```

Comment:
Yeah, I suppose I agree, though it occurs to me that the issue is that we don't have a way to append to a buffer. Perhaps this function should act like `setline`, but we should also have a `buffer_append` function to avoid race conditions.

Those that can be in a different PR.